src/tests/acp-agent.test.ts

```diff
@@ -769,6 +769,22 @@ describe("tool conversions", () => {
   });
 });
 
+describe("setSessionMode bypass permissions error handling", () => {
```

Comment:
I'm not sure this test adds much. I can see how it was helpful to learn the behavior, but I feel we should pull this into a helper function, or just remove the test (totally fine with the latter by the way: it helped you learn, but we don't need to spend CI cycles testing it forever)